### PR TITLE
GG-17373 Mock web sockets calls for E2E tests

### PR DESCRIPTION
**Pull request by Klaster1**
@nva what do you think of WS mock API and mock methods/data for Agent tasks?

### CODE REVIEW

File: modules/web-console/frontend/app/modules/agent/AgentManager.service.ts [hunk 1]
```diff
@@ -477,10 +458,13 @@ export default class AgentManager {
 
                             const useBigIntJson = taskId.startsWith('query');
 
-                            return this.pool.postMessage({payload: res.data, useBigIntJson});
+                            const response = await this.pool.postMessage({payload: res.data, useBigIntJson});
+                            return response;
                         }
 
-                        return res;
+                        if (NON_ZIPPED_TASKS.has(params.taskId))
```
Comment (nva) [inline modules/web-console/frontend/app/modules/agent/AgentManager.service.ts]: This code is not needed: AgentManager.drivers, AgentManager.schemas, AgentManager.schemas.tables call AgentManager._sendToAgent directly

**Comment by nva** [comment]
Please resolve conflict.

File: modules/web-console/frontend/app/modules/agent/AgentManager.service.ts [hunk 2]
```diff
@@ -577,12 +561,9 @@ export default class AgentManager {
...
     }
 
-    /**
-     * @param {string} cacheName Cache name.
-     */
-    cacheNodes(cacheName) {
+    cacheNodes(cacheName: string) {
         if (this.available(IGNITE_2_0))
-            return this.visorTask('cacheNodesTaskX2', null, cacheName);
+            return this.visorTask<AgentTypes.CacheNodesTaskX2Response>('cacheNodesTaskX2', null, cacheName);
```
[the remark this answers is not in the document]
Comment (Klaster1) [reply modules/web-console/frontend/app/modules/agent/AgentManager.service.ts]: Good point, I've renamed the type to CacheNodesTaskResponse and use it for both calls.

File: modules/web-console/frontend/app/types/Agent.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+export type AgentsStatResponse = {
+	count: number,
+	hasDemo: boolean,
+	clusters: ClusterStats[]
+}
+
+export type CacheNamesCollectorTaskResponse = {
+	caches: {[cacheName: string]: string},
+	groups: string[],
+	protocolVersion: 1
```
[the remark this answers is not in the document]
Comment (Klaster1) [reply modules/web-console/frontend/app/types/Agent.ts]: Removed.

File: modules/web-console/frontend/app/types/Agent.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+export type CacheNodesTaskX2Response = string[]
+
+export type VisorQueryResult =
+{
+	error: null,
+	result: {
+		columns: null | {
+			fieldName: string,
+			fieldTypeName: string,
+			protocolVersion: 1,
```
[the remark this answers is not in the document]
Comment (Klaster1) [reply modules/web-console/frontend/app/types/Agent.ts]: Removed.

File: modules/web-console/frontend/app/modules/agent/AgentManager.service.ts (hunk 1 above)
Comment (Klaster1) [reply modules/web-console/frontend/app/modules/agent/AgentManager.service.ts]: Done.

**Pull request closed without merging**